### PR TITLE
Support tpcds benchmark

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Support tpcds benchmark in `dev/kyuubi-tpcds` module.

Add a `README.md` in `dev/kyuubi-tpcds` module to show how to use.

The mian code is from [databricks-spark-sql-perf](https://github.com/databricks/spark-sql-perf)

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/TPCDS_2_4_Queries.scala [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    "ss_max")
+
+  val tpcds2_4Queries = queryNames.map { queryName =>
+    val queryContent: String = IOUtils.toString(
+      getClass().getClassLoader().getResourceAsStream(s"tpcds_2_4/$queryName.sql"))
+    Query(
+      queryName + "-v2.4",
+      queryContent,
+      description = "TPCDS 2.4 Query",
+      executionMode = CollectResults)
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/TPCDS_2_4_Queries.scala:142]: expose `executionMode` to args?

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/Query.scala [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      // lazily evaluated above, so below we will count only execution time.
+      var result: Option[Long] = None
+      val executionTime = measureTimeMs {
+        executionMode match {
+          case ExecutionMode.CollectResults => dataFrame.collect()
+          case ExecutionMode.ForeachResults => dataFrame.foreach { _ => (): Unit }
+          case ExecutionMode.WriteParquet(location) =>
+            dataFrame.write.parquet(s"$location/$name.parquet")
+          case ExecutionMode.HashResults =>
+            // SELECT SUM(CRC32(CONCAT_WS(", ", *))) FROM (benchmark query)
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/Query.scala:128]: How about `CRC32(CONCAT_WS(", ", *))` => `hash(*)`?

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/Query.scala [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      // to scala.
+      // Note: queryExecution.{logical, analyzed, optimizedPlan, executedPlan} has been already
+      // lazily evaluated above, so below we will count only execution time.
+      var result: Option[Long] = None
+      val executionTime = measureTimeMs {
+        executionMode match {
+          case ExecutionMode.CollectResults => dataFrame.collect()
+          case ExecutionMode.ForeachResults => dataFrame.foreach { _ => (): Unit }
+          case ExecutionMode.WriteParquet(location) =>
+            dataFrame.write.parquet(s"$location/$name.parquet")
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/Query.scala:126]: `mode.(Overwrite)`

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/ExecutionMode.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  case object ForeachResults extends ExecutionMode {
+    override def toString: String = "foreach"
+  }
+
+  /** Benchmark run by saving the output of each query as a parquet file. */
+  case class WriteParquet(location: String) extends ExecutionMode {
+    override def toString: String = "saveToParquet"
+  }
+
+  /**
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/ExecutionMode.scala:42]: Add `WriteNoop`
```
df.write.format("noop").mode(Overwrite).save()
```

File: dev/kyuubi-tpcds/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+package jar with following command:
+`./build/mvn install -DskipTests -Ptpcds -pl dev/kyuubi-tpcds -am`
+
+## data generator 
+Run following command to generate 10GB data with new database `tpcds_sf10`.
+
+```shell
+$SPARK_HOME/bin/spark-submit \
+  --conf spark.sql.tpcds.scale.factor=10 \
+  --conf spark.sql.tpcds.database=tpcds_sf10 \
```
Comment (pan3793) [inline dev/kyuubi-tpcds/README.md:32]: Unify the arg parser, migrate to scopt

File: dev/kyuubi-tpcds/pom.xml [hunk 6]
```diff
@@ -43,6 +43,22 @@
             <artifactId>spark-sql_${scala.binary.version}</artifactId>
             <scope>provided</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.github.scopt</groupId>
+            <artifactId>scopt_2.12</artifactId>
```
Comment (pan3793) [inline dev/kyuubi-tpcds/pom.xml:49]: `${scala_binary_version}`

File: dev/kyuubi-tpcds/src/main/resources/tpcds_2_4/q1.sql [hunk 7]
```diff
@@ -0,0 +1,19 @@
+--q1.sql--
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/resources/tpcds_2_4/q1.sql:1]: Add ASF header to SQL files

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/ExecutionMode.scala (hunk 4 above)
Comment (ulysses-you) [reply dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/ExecutionMode.scala:42]: we can do this in a new PR if we really need them

File: dev/kyuubi-tpcds/pom.xml (hunk 6 above)
Comment (ulysses-you) [reply dev/kyuubi-tpcds/pom.xml]: fixed

File: dev/kyuubi-tpcds/README.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+package jar with following command:
+`./build/mvn install -DskipTests -Ptpcds -pl dev/kyuubi-tpcds -am`
+
+## data generator 
+Run following command to generate 10GB data with new database `tpcds_sf10`.
+
+```shell
+$SPARK_HOME/bin/spark-submit \
+  --conf spark.sql.tpcds.scale.factor=10 \
```
Comment (yaooqinn) [inline dev/kyuubi-tpcds/README.md:31]: can we have a table to describe all supported configs?

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+        System.exit(1)
+    }
+  }
+
+  def run(config: RunConfig): Unit = {
+    val conf = new SparkConf()
+      .setAppName(config.benchmarkName)
+
+    val sparkSession = SparkSession.builder.config(conf).enableHiveSupport().getOrCreate()
+    val sqlContext = sparkSession.sqlContext
```
Comment (yaooqinn) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:76]: why we have to keep the legacy `sqlContext`?

**Review by pan3793 — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**